base/sort.jl

```diff
@@ -1298,6 +1298,53 @@ function sort!(A::AbstractArray;
     end
     A
 end
+"""
+    sortperm(A; dims::Integer, alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
+
+    Return a permutation vector `I` that puts `v[I]` in sorted order along the given dimension. The order is specified
+    using the same keywords as [`sort!`](@ref). The permutation is guaranteed to be stable even
+    if the sorting algorithm is unstable, meaning that indices of equal elements appear in
+    ascending order.
+
+    To sort slices of an array, refer to [`sortslices`](@ref).
+
+    # Examples
+    ```jldoctest
+    julia> A = [4 3; 1 2]
+    2×2 Matrix{Int64}:
+     4  3
+     1  2
+    
+    julia> sortperm(A, dims = 1)
+    2×2 Matrix{Int64}:
+     2  4
+     1  3
+    
+    julia> sortperm(A, dims = 2)
+    2×2 Matrix{Int64}:
+     3  1
+     2  4
+```
+"""
+function sortperm(A::AbstractArray; dims::Integer,
```

Comment:
`dims` is not an ideal name for a parameter that can only specify a single dimension, but we are already committed to that interface choice. Internally, we can still do what is done for `sort(A::AbstractArray; dims::Integer)` which is assign `dim=dims` at the start of the function. (line 1228)

Reply:
Or not. If we only end up using dims once or twice that might not be helpful.